docs/core/deploying/trimming-options.md

```diff
@@ -0,0 +1,125 @@
+
+```xml
+<ItemGroup>
+  <TrimmerRootAssembly Include="MyAssembly" />
+</ItemGroup>
+```
+
+## Root descriptors
+
+Another way to specify roots for analysis is using an XML file that uses the linker [descriptor format](https://github.com/mono/linker/blob/master/docs/data-formats.md#descriptor-format). This lets you root specific members instead of a whole assembly.
+
+```xml
+<ItemGroup>
+  <TrimmerRootDescriptor Include="MyRoots.xml" />
+</ItemGroup>
+```
+
+For example, `MyRoots.xml` might root a specific method that is dynamically accessed by the application:
+
+```xml
+<linker>
+  <assembly fullname="MyAssembly">
+    <type fullname="MyAssembly.MyClass">
+      <method signature="System.Void DynamicallyAccessedMethod()" />
+    </type>
+  </assembly>
+</linker>
+```
+
+## Analysis warnings
+
+Trimming will remove IL that is not statically reachable. Apps that use reflection or other patterns that create dynamic dependencies may be broken by trimming. To warn about such patterns:
+
+- `<SuppressTrimAnalysisWarnings>false</SuppressTrimAnalysisWarnings>`
+
+    Enable trim analysis warnings.
+
+This will include warnings about the entire app, including your own code, library code, and framework code.
+
+## Warning versions
+
+Trim analysis respects the [`AnalysisLevel`](../project-sdk/msbuild-props.md#AnalysisLevel) property that controls the version of analysis warnings across the SDK. There is another property that controls the version of trim analysis warnings independently (similar to `WarningLevel` for the compiler):
+
+- `<ILLinkWarningLevel>5</ILLinkWarningLevel>`
+
+    Emit only warnings of the given level or lower. This can be `9999` to include all warning versions.
+
+## Suppressing warnings
+
+Individual [warning codes](https://github.com/mono/linker/blob/master/docs/error-codes.md#warning-codes) can be suppressed using the usual MSBuild properties respected by the toolchain, including `NoWarn`, `WarningsAsErrors`, `WarningsNotAsErrors`, and `TreatWarningsAsErrors`. There is an additional option that controls the ILLink warn-as-error behavior independently:
+
+- `<ILLinkTreatWarningsAsErrors>false</ILLinkTreatWarningsAsErrors>`
+
+    Don't treat ILLink warnings as errors. This may be useful to avoid turning trim analysis warnings into errors when treating compiler warnings as errors globally.
+
+## Removing symbols
+
+Symbols will normally be trimmed to match the trimmed assemblies. You can also remove all symbols:
+
+- `<TrimmerRemoveSymbols>true</TrimmerRemoveSymbols>`
+
+    Remove symbols from the trimmed application, including embedded PDBs and separate PDB files. This applies to both the application code and any dependencies that come with symbols.
+
+The SDK also makes it possible to disable debugger support using the property `DebuggerSupport`. When debugger support is disabled, trimming will remove symbols automatically (`TrimmerRemoveSymbols` will default to true).
```

Comment:
We have them listed in the repo: https://github.com/dotnet/runtime/blob/master/docs/workflow/trimming/feature-switches.md.

I think it would be good to get them in the docs somewhere, but I'm not sure where. Would this file `trimming-options.md` be a decent place for these?

Reply:
Oh, nice! I think this file would be an ok place to put the list - I will get them added. I also found these existing docs:
- A [runtimeconfig overview](https://docs.microsoft.com/en-us/dotnet/core/run-time-config/) links to [runtimeconfig properties](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#run-time-configuration-properties), which lists the threading/JIT/GC (non-trimmable) options and InvariantGlobalization, with a link to [InvariantGlobalization](https://docs.microsoft.com/en-us/dotnet/core/run-time-config/globalization#invariant-mode)
- Breaking changes docs for [UTF-7](https://docs.microsoft.com/en-us/dotnet/core/compatibility/corefx#utf-7-code-paths-are-obsolete) and [BinaryFormatter](https://docs.microsoft.com/en-us/dotnet/core/compatibility/corefx#binaryformatter-serialization-methods-are-obsolete-and-prohibited-in-aspnet-apps)

That leaves
- DebuggerSupport
- EventSourceSupport
- HttpActivityPropagationSupport
- UseSystemResourceKeys. 

Do you think any of those are also worth documenting independently of trim support?

Reply:
> Do you think any of those are also worth documenting independently of trim support?

At this point, I think those 4 are only useful in the context of trimming support.